pytorch_lightning/metrics/functional/psnr.py

```diff
@@ -1,28 +1,36 @@
-from typing import Optional, Tuple
+from typing import Optional, Tuple, Union
 
 import torch
 
-from pytorch_lightning.utilities import rank_zero_warn
+from pytorch_lightning.metrics import utils
 
 
-def _psnr_compute(
-    sum_squared_error: torch.Tensor,
-    n_obs: int,
+def _psnr(
```

Comment:
In this commit the `PSNR.update` took the log of mean-squared-error and save _PSNR_ scores as states when `dim is not None`. When `dim is None`, it took the log in `PSNR.compute`. So the log calculation was extracted as `_psnr`.

I update the `PSNR` so that the log calculation is always in `PSNR.compute` no matter the value of `dim`. And the `_psnr` is removed.

Reply:
nice :]